ragna/deploy/_cli/core.py

```diff
@@ -181,3 +188,175 @@ def wait_for_api() -> None:
         ui_app(config=config, open_browser=open_browser).serve()  # type: ignore[no-untyped-call]
     finally:
         shutdown_api()
+
+
+@corpus_app.command(help="Ingest some documents into a given corpus.")
+def ingest(
+    documents: list[Path],
+    metadata_fields: Optional[Path] = typer.Option(
+        None,
+        help="JSON file that contains mappings from document name "
+        "to metadata fields associated with a document.",
+    ),
+    corpus_name: Optional[str] = typer.Option(
+        None, help="Name of the corpus to ingest the documents into."
+    ),
+    config: ConfigOption = "./ragna.toml",  # type: ignore[assignment]
+    user: Optional[str] = typer.Option(
+        None, help="User to link the documents to in the ragna database."
+    ),
+    report_failures: bool = typer.Option(
+        False, help="Output to STDERR the documents that failed to be ingested."
+    ),
+    ignore_log: bool = typer.Option(
+        False, help="Ignore the log file and re-ingest all documents."
+    ),
+) -> None:
+    try:
+        document_factory = getattr(config.document, "from_path")
+    except AttributeError:
+        raise typer.BadParameter(
+            f"{config.document.__name__} does not support creating documents from a path. "
+            "Please implement a `from_path` method."
+        )
+
+    try:
+        make_session = database.get_sessionmaker(config.api.database_url)
+    except Exception:
+        raise typer.BadParameter(
+            f"Could not connect to the database: {config.api.database_url}"
+        )
+
+    if metadata_fields:
```

Comment:
This file should contain a mapping from individual filepaths to a dictionary of metadata fields for each filepath.